Themes/default/GenericControls.template.php

```diff
@@ -154,7 +154,25 @@ function template_control_richedit_buttons($editor_id)
 	elseif ($context['show_spellchecking'])
 		$tempTab++;
 
-	$tempTab++;
+	// add a cancel button for these actions
+	if(isset($_REQUEST['action']) && in_array($_REQUEST['action'], array('post', 'calendar')))
+	{
+		$cancelLink = '';
+		if(isset($_REQUEST['msg']) || isset($_REQUEST['last_msg']))
+			$cancelLink = $scripturl . '?topic='. $_REQUEST['topic'] .'.msg'. (isset($_REQUEST['msg']) ? $_REQUEST['msg'] .'#msg'. $_REQUEST['msg'] : $_REQUEST['last_msg'] .'#msg'. $_REQUEST['last_msg']);
+		else if(isset($_REQUEST['board']))
+			$cancelLink = $scripturl . '?board='. $_REQUEST['board']; 
+		else if($_REQUEST['action'] == 'calendar')
+			$cancelLink = $scripturl . '?action='. $_REQUEST['action'];
+
+		if(!empty($cancelLink))
+		{
+			$tempTab++;
+			echo '
+		<input type="button" value="'. $txt['modify_cancel'] .'" name="'. $txt['modify_cancel'] .'" tabindex="', $tempTab, '" onclick="window.location.href=\'', $cancelLink ,'\'" accesskey="c" class="button_submit">';
```

Comment:
old code ;-)

The correct "cancel button" is define in the Subs-Editor.php just before
`if (empty($context['bbc_tags']))`
and looks as:

`	// setup the cancel button
	$isQuickReply = $editorOptions['id'] == 'quickReply';
	$isAction = isset($_GET['action']) && in_array($_GET['action'], array('post', 'post2', 'calendar')) ? $_GET['action'] : false;
	if($isQuickReply || $isAction)
	{
		if($isQuickReply || in_array($isAction, array('post', 'post2')) || ($isAction == 'calendar'	&& isset($_GET['sa']) && $_GET['sa'] == 'post'))
		{
			if(!empty($modSettings['sef_enabled']))
				$cururl = pmxsef_query(str_replace($boardurl, '', isset($_SERVER['HTTP_REFERER']) ? $_SERVER['HTTP_REFERER'] : $_SERVER['REQUEST_URL']));
			else
				$cururl = parse_url(rawurldecode(str_replace($scripturl .'?', '', isset($_SERVER['HTTP_REFERER']) ? $_SERVER['HTTP_REFERER'] : $_SERVER['REQUEST_URL'])));

			if(!isset($cururl['action']) || (isset($cururl['action']) && !in_array($cururl['action'], array('post', 'post2')) && (!isset($cururl['sa']) || (isset($cururl['sa']) && $cururl['sa'] !== 'post')) && !isset($cururl['id_draft'])))
			{
				$fromWhere = $isQuickReply ? $_SERVER['REQUEST_URL'] : $_SERVER['HTTP_REFERER'];
				if(strpos($fromWhere, 'showposts') == false)
					preg_match('/(msg[0-9]+)/', $fromWhere, $fragment);
				$context['controls']['richedit'][$editorOptions['id']]['cancel_link'] = $_SESSION['cancel_link'] = $fromWhere . (isset($fragment[0]) ? '#'. $fragment[0] : '#top');
			}
			else
				$context['controls']['richedit'][$editorOptions['id']]['cancel_link'] = $_SESSION['cancel_link'];
		}
		elseif($isAction == 'post2' || isset($_GET['id_draft']))
			$context['controls']['richedit'][$editorOptions['id']]['cancel_link'] = $_SESSION['cancel_link'];
	}

	if (empty($context['bbc_tags']))
`